src/EnergyPlus/DaylightingManager.cc

```diff
@@ -679,120 +658,114 @@ void CalcDayltgCoefficients(EnergyPlusData &state)
-                                IHR, ISlatAngle, static_cast<int>(DataDaylighting::SkyType::Overcast), refPtNum, loop);
-
-                            // write daylight factors - 4 sky types for each daylight ref point
+                    // loop over each slat angle
+                    for (ISlatAngle = 1; ISlatAngle <= ISA; ++ISlatAngle) {
+                        if (ISlatAngle == 1) {
+                            // base window without shades, screens, or blinds
+                            print(state.files.dfs,
+                                  "{},{},{},Base Window\n",
+                                  state.dataEnvrn->CurMnDy,
+                                  state.dataHeatBal->Zone(thisDaylightControl.zoneIndex).Name,
+                                  state.dataSurface->Surface(windowSurfNum).Name);
+                        } else if (ISlatAngle == 2 && ISA == 2) {
+                            // window shade or blind with fixed slat angle
+                            print(state.files.dfs,
+                                  "{},{},{},Blind or Slat Applied\n",
+                                  state.dataEnvrn->CurMnDy,
+                                  state.dataHeatBal->Zone(thisDaylightControl.zoneIndex).Name,
+                                  state.dataSurface->Surface(windowSurfNum).Name);
+                        } else {
+                            // blind with variable slat angle
+                            SlatAngle = 180.0 / double(MaxSlatAngs - 1) * double(ISlatAngle - 2);
                             print(state.files.dfs,
-                                  "{},{},{:.5R},{:.5R},{:.5R},{:.5R}\n",
-                                  IHR,
-                                  state.dataDaylightingData->DaylRefPt(state.dataDaylightingData->ZoneDaylight(ZoneNum).DaylRefPtNum(refPtNum)).Name,
-                                  DFClrSky,
-                                  DFClrTbSky,
-                                  DFIntSky,
-                                  DFOcSky);
-                        } // Reference Point loop
-                    }     // hour loop
-                }         // slat angle loop
-            }
-        } // exterior windows in zone loop
-    }     // zone loop
+                                  "{},{},{},{:.1R}\n",
+                                  state.dataEnvrn->CurMnDy,
+                                  state.dataHeatBal->Zone(thisDaylightControl.zoneIndex).Name,
+                                  state.dataSurface->Surface(windowSurfNum).Name,
+                                  SlatAngle);
+                        }
+
+                        for (int IHR = 1; IHR <= 24; ++IHR) {
+                            // For each Daylight Reference Point
+                            for (int refPtNum = 1; refPtNum <= thisDaylightControl.TotalDaylRefPoints; ++refPtNum) {
+                                DFClrSky = thisDaylightControl.DaylIllFacSky(
```

Comment:
These arrays are just crazy and will have to be revisited at some point in the future.